airflow/ui/src/pages/Variables/AddVariableForm.tsx

```diff
@@ -0,0 +1,119 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { Box, HStack, Input, Spacer, Text, Textarea } from "@chakra-ui/react";
+import { useEffect, useMemo } from "react";
+import { Controller, useForm } from "react-hook-form";
+import { FiSave } from "react-icons/fi";
+
+import { ErrorAlert } from "src/components/ErrorAlert";
+import { Button } from "src/components/ui";
+import { useAddVariable } from "src/queries/useAddVariable";
+
+export type AddVariableBody = {
+  description: string | undefined;
+  key: string;
+  value: string;
+};
+
+type AddVariableFormProps = {
+  onClose: () => void;
+};
+
+const AddVariableForm: React.FC<AddVariableFormProps> = ({ onClose }) => {
+  const { addVariable, error, isPending } = useAddVariable(onClose);
+  const addVariableParams: AddVariableBody = useMemo(
```

Comment:
Why a memo here ? That’s really expensive on the first render, and does not seem to prevent heavy computing.

Reply:
Actually yes, Memo is not required, I was facing error for it `The 'addVariableParams' object makes the dependencies of useEffect Hook (at line 61) change on every render. ` it was suggesting me to use memo, but I have changed the approach in this way memo is not used.